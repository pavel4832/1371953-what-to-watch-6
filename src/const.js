export const RATING_STARS = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];

export const COLUMN_QUANTITY = 2;

export const PRECISION_RATING = 2;

export const TIMEOUT_PREVIEW = 1000;

export const CONTENT_TYPE = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `reviews`
};

export const POSTER_SIZE = {
  WIDTH: `280`,
  HEIGHT: `175`
};

export const COUNT_CARD = {
  SAME: 4,
  MAIN_PER_STEP: 8,
  ACTIVE: 0
};

export const FILTER_TYPE = {
  ALL_GENRE: `All genre`,
  COMEDIES: `Comedies`,
  CRIME: `Crime`,
  DOCUMENTARY: `Documentary`,
  DRAMAS: `Dramas`,
  HORROR: `Horror`,
  KIDS_FAMILY: `kids & Family`,
  ROMANCE: `Romance`,
  SCI_FI: `Sci-Fi`,
  THRILLERS: `Thrillers`
};

export const MAX_FILTERS = 9;

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const AppRoute = {
  ROOT: `/`,
  LOGIN: `/login`,
  MY_LIST: `/mylist`,
  FILMS: `/films`,
  PLAYER: `/player/:id`,
};

export const APIRoute = {
  FILMS: `/films`,
  PROMO: `/films/promo`,
  COMMENTS: `/comments`,
  LOGIN: `/login`,
  ERROR: `/error`
};

export const EMAIL_REGEXP = /^(([^<>()[\].,;:\s@"]+(\.[^<>()[\].,;:\s@"]+)*)|(".+"))@(([^<>()[\].,;:\s@"]+\.)+[^<>()[\].,;:\s@"]{2,})$/iu;
