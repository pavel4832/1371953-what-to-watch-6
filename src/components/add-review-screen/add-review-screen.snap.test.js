import React from 'react';
import {Provider} from 'react-redux';
import {render} from '@testing-library/react';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import AddReviewScreen from './add-review-screen';
import configureStore from 'redux-mock-store';
import {AuthorizationStatus} from "../../const";

const mockStore = configureStore({});

it(`Should AddReviewScreen render correctly`, () => {
  const id = 1;
  const history = createMemoryHistory();
  const store = mockStore({
    USER: {authorizationStatus: AuthorizationStatus.AUTH}
  });
  const {container} = render(
      <Provider store={store}>
        <Router history={history}>
          <AddReviewScreen id={id} />
        </Router>
      </Provider>
  );
  expect(container).toMatchSnapshot();
});
