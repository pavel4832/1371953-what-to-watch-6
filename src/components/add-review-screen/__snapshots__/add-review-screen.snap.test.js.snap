// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should AddReviewScreen render correctly 1`] = `
<div>
  <section
    class="movie-card movie-card--full"
  >
    <div
      class="movie-card__header"
    >
      <div
        class="movie-card__bg"
      >
        <img
          alt="The Grand Budapest Hotel"
          src="img/bg-the-grand-budapest-hotel.jpg"
        />
      </div>
      <h1
        class="visually-hidden"
      >
        WTW
      </h1>
      <header
        class="page-header movie-card__head "
      >
        <div
          class="logo"
        >
          <a
            class="logo__link"
            href="/"
          >
            <span
              class="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              class="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              class="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <nav
          class="breadcrumbs"
        >
          <ul
            class="breadcrumbs__list"
          >
            <li
              class="breadcrumbs__item"
            >
              <a
                class="breadcrumbs__link"
                href="/films/1"
              >
                The Grand Budapest Hotel
              </a>
            </li>
            <li
              class="breadcrumbs__item"
            >
              <a
                class="breadcrumbs__link"
                href="#"
              >
                Add review
              </a>
            </li>
          </ul>
        </nav>
        <h1
          class="page-title user-page__title"
          style="display: none;"
        />
        <div
          class="user-block"
        >
          <div
            class="user-block__avatar"
          >
            <a
              href="/mylist"
            >
              <img
                alt="User avatar"
                height="63"
                src="img/avatar.jpg"
                width="63"
              />
            </a>
          </div>
          <a
            class="user-block__link"
            href="/"
          >
            Log out
          </a>
        </div>
      </header>
      <div
        class="movie-card__poster movie-card__poster--small"
      >
        <img
          alt="The Grand Budapest Hotel-poster"
          height="198"
          src="img/the-grand-budapest-hotel-poster.jpg"
          width="132"
        />
      </div>
    </div>
    <div
      class="add-review"
    >
      <form
        action="#"
        class="add-review__form"
      >
        <div
          class="rating"
        >
          <div
            class="rating__stars"
          >
            <input
              class="rating__input"
              data-testid="rating"
              id="star-1"
              name="rating"
              type="radio"
              value="1"
            />
            <label
              class="rating__label"
              for="star-1"
            >
              Rating 
              1
            </label>
            <input
              class="rating__input"
              data-testid="rating"
              id="star-2"
              name="rating"
              type="radio"
              value="2"
            />
            <label
              class="rating__label"
              for="star-2"
            >
              Rating 
              2
            </label>
            <input
              class="rating__input"
              data-testid="rating"
              id="star-3"
              name="rating"
              type="radio"
              value="3"
            />
            <label
              class="rating__label"
              for="star-3"
            >
              Rating 
              3
            </label>
            <input
              class="rating__input"
              data-testid="rating"
              id="star-4"
              name="rating"
              type="radio"
              value="4"
            />
            <label
              class="rating__label"
              for="star-4"
            >
              Rating 
              4
            </label>
            <input
              class="rating__input"
              data-testid="rating"
              id="star-5"
              name="rating"
              type="radio"
              value="5"
            />
            <label
              class="rating__label"
              for="star-5"
            >
              Rating 
              5
            </label>
            <input
              class="rating__input"
              data-testid="rating"
              id="star-6"
              name="rating"
              type="radio"
              value="6"
            />
            <label
              class="rating__label"
              for="star-6"
            >
              Rating 
              6
            </label>
            <input
              class="rating__input"
              data-testid="rating"
              id="star-7"
              name="rating"
              type="radio"
              value="7"
            />
            <label
              class="rating__label"
              for="star-7"
            >
              Rating 
              7
            </label>
            <input
              class="rating__input"
              data-testid="rating"
              id="star-8"
              name="rating"
              type="radio"
              value="8"
            />
            <label
              class="rating__label"
              for="star-8"
            >
              Rating 
              8
            </label>
            <input
              class="rating__input"
              data-testid="rating"
              id="star-9"
              name="rating"
              type="radio"
              value="9"
            />
            <label
              class="rating__label"
              for="star-9"
            >
              Rating 
              9
            </label>
            <input
              checked=""
              class="rating__input"
              data-testid="rating"
              id="star-10"
              name="rating"
              type="radio"
              value="10"
            />
            <label
              class="rating__label"
              for="star-10"
            >
              Rating 
              10
            </label>
          </div>
        </div>
        <div
          class="add-review__text"
        >
          <textarea
            class="add-review__textarea"
            data-testid="comment"
            id="review-text"
            maxlength="400"
            minlength="50"
            name="review-text"
            placeholder="Review text"
          />
          <div
            class="add-review__submit"
          >
            <button
              class="add-review__btn"
              disabled=""
              type="submit"
            >
              Post
            </button>
          </div>
        </div>
      </form>
    </div>
  </section>
</div>
`;
